### DISCUSSION TITLE
UploadFile content_type shows the wrong mimetype for .ts files

### DISCUSSION THREAD

**Discussion by SionAbes** [fastapi/fastapi#10305] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from app.api.model import video as model
from fastapi import APIRouter, Form, UploadFile, status

router = APIRouter(
    prefix="/video",
    tags=["Video"],
)


@router.post("/", status_code=status.HTTP_201_CREATED)
def upload_video(
    material_id: Annotated[str, Form()],
    ingested_from: Annotated[str, Form()],
    video_file: UploadFile,
) -> model.VideoRecord:

    mimetype_parent, mimetype_child = video_file.content_type.split("/")
    if mimetype_parent != "video":
        raise ValueError(
            f"File of type {video_file.content_type} is not an accepted video mimetype!"
        )

...

import shutil
from pathlib import Path

from app.main import app
from starlette import status
import pytest

def test_upload_video(client):
    _test_upload_file = Path("tests/data/", "test_file.ts")
    _files = {
        "video_file": _test_upload_file.open("rb"),
    }
    payload = {"material_id": "_material_id", "ingested_from": "tests"}

    url = app.url_path_for("upload_video")
    response = client.post(url, files=_files, data=payload)
```

### Description

I'm currently building an API as access to a huge video and audio store, and want to validate the mimetype of the uploaded file before going forward with the business logic of my application.

I've tested with an `.mp4` file, whereby `UploadFile.content_type` displays the correct mimetype `video/mp4`, but I'm having trouble with `.ts` files. It should display `video/mp2t` as mentioned in the [standards](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types), but I instead get `application/octet-stream` leading to my ValueError raising.

Any idea if this is intended functionality?

Thank you for you help!
SionAbes

Edits: spelling, grammar and formatting!

### Operating System

macOS

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

Python 3.11.3

**Comment by jgould22** [comment, marked as answer]
You are not specifying the content type in your request.

When you leave that out HTTPX will try and guess the type as you can see here https://github.com/encode/httpx/blob/master/httpx/_utils.py#L293

It uses pythons mimetypes module https://docs.python.org/3/library/mimetypes.html

Pythons mimetypes module uses some files from the system you are on to guess the extension.

On my system(linux) one of the files it checks is `/etc/mime.types`.

If you add something like `video/mp2t				ts` to the `/etc/mime.types` file in your applications environment HTTPX will likely guess the type correctly (though this may vary on MacOS so you will have to do some research).

You could also just set the content-type header yourself for that post request.

**Comment by alex-pobeditel-2004** [comment]
Hello! I'd recommend you to use python-magic that uses libmagic which is a real killer in the type guessing:
https://pypi.org/project/python-magic
You must never believe to content type specified by your clients, and python`s internal type guesser is not so reliable, too.
